cf/mixin/fielddomain.py

```diff
@@ -2408,6 +2408,110 @@ def set_construct(
+
+                {{default Exception}}
+
+            {{filter_kwargs: optional}}
+
+                .. versionadded:: NEXTVERSION
+
+        :Returns:
+
+                The removed metadata construct.
+
+        **Examples**
+
+        >>> f = {{package}}.example_field(0)
+        >>> print(f)
+        Field: specific_humidity (ncvar%q)
+        ----------------------------------
+        Data            : specific_humidity(latitude(5), longitude(8)) 1
+        Cell methods    : area: mean
+        Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
+                        : longitude(8) = [22.5, ..., 337.5] degrees_east
+                        : time(1) = [2019-01-01 00:00:00]
+        >>> f.del_construct('time')
+        <{{repr}}DimensionCoordinate: time(1) days since 2018-12-01 >
+        >>> f.del_construct('time')
+        Traceback (most recent call last):
+            ...
+        ValueError: Can't find unique construct to remove
+        >>> f.del_construct('time', default='No time')
+        'No time'
+        >>> f.del_construct('dimensioncoordinate1')
+        <{{repr}}DimensionCoordinate: longitude(8) degrees_east>
+        >>> print(f)
+        Field: specific_humidity (ncvar%q)
+        ----------------------------------
+        Data            : specific_humidity(latitude(5), ncdim%lon(8)) 1
+        Cell methods    : area: mean
+        Dimension coords: latitude(5) = [-75.0, ..., 75.0] degrees_north
+
+        """
+        # Need to re-define to overload this method since cfdm doesn't
+        # have the concept of cyclic axes, so have to update the
+        # register of cyclic axes when we delete a construct in cf.
+
+        # Get the relevant key first because it will be lost upon deletion.
+        key = self.construct_key(*identity, default=None, **filter_kwargs)
+        # Copy since should never change value of _cyclic attribute in-place
+        cyclic_axes = self._cyclic.copy()
+
+        deld_construct = super().del_construct(
+            *identity, default=default, **filter_kwargs
+        )
+
+        # If the construct deleted was a cyclic axes, remove it from the set
+        # of stored cyclic axes, to update that appropriately. Do this
+        # afterwards because the deletion might not be successful and don't
+        # want to update the cyclic() set unless we know the deletion occurred.
+        if key in cyclic_axes:
+            # The below is to test that the construct was successfully deleted
+            if isinstance(default, Exception) or (
+                not isinstance(default, Exception)
+                and deld_construct != default
+            ):
+                cyclic_axes.remove(key)
+                self._cyclic = cyclic_axes
+                logger.info(
+                    "Deleted a construct that corresponds to a cyclic axis "
+                    f"({key}), so it was removed from the cyclic() axes set."
+                )
```

Comment:
Ultimately, no. I do think it would be nice as verbose information to at least add "Deleted a construct that corresponds to a cyclic axis." i.e. the first line, but potentially as a `logger.detail` or even `debug` level if you think it is a bit much to tell the user on `info` level?

Reply:
Hi Sadie - I'm afraid that I think it's too much at any level :) To me this is just basic functionality that we shouldn't make slower with logging calls.